File: reportservice/src/main/java/dev/sverdlov/reportservice/dto/ReportInfoResponseDto.java
package dev.sverdlov.reportservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.UUID;

public class ReportInfoResponseDto {
    @JsonProperty("query")
    private final UUID query;

    @JsonProperty("percent")
    private final int percent;

    @JsonProperty("result")
    private final ReportResultDto result;

    public ReportInfoResponseDto(UUID query, int percent, ReportResultDto result) {
        this.query = query;
        this.percent = percent;
        this.result = result;
    }

    public UUID getQuery() {
        return query;
    }

    public int getPercent() {
        return percent;
    }

    public ReportResultDto getResult() {
        return result;
    }
}

File: ReportServiceUsingCsharp/Dto/ReportInfoResponseDto.cs
using System.Text.Json.Serialization;

namespace ReportServiceUsingCsharp.Dto;

public class ReportInfoResponseDto
{
    [JsonPropertyName("query")]
    public Guid Query { get; }

    [JsonPropertyName("percent")]
    public int Percent { get; }

    [JsonPropertyName("result")]
    public ReportResultDto? Result { get; }

    public ReportInfoResponseDto(Guid query, int percent, ReportResultDto? result)
    {
        Query = query;
        Percent = percent;
        Result = result;
    }
}